_drafts/2023-04-15-functional-error-handling-in-kotlin.md

```diff
@@ -0,0 +1,811 @@
+---
+title: "Functional Error Handling in Kotlin: Part 1"
+date: 2023-04-27
+header:
+    image: "/images/blog cover.jpg"
+tags: [kotlin]
+excerpt: "Whether we develop using an object-oriented or functional approach, we always have the problem of handling errors. Kotlin offers a lot of different methods to do it. Here, we'll focus on strategies that don't manage the cause of the error, i.e., nullable types and Arrow `Option` types."
+---
+
+The Kotlin language is a multi-paradigm, general-purpose programming language. Whether we develop using an object-oriented or functional approach, we always have the problem of handling errors. Kotlin offers a lot of different methods to handle errors. Still, this article will focus on the functional approaches and introduce the Arrow library. This article is the first part of a series. We'll focus on strategies that don't manage the cause of the error, i.e., nullable types and Arrow `Option` types. So, without further ado, let's get started.
```

Comment:
> We'll focus on strategies that don't manage the cause of the error, i.e., nullable types and Arrow `Option` types

I know what you mean, but I had to read it multiple times to understand. I'd suggest: "We'll focus on strategies that deal with the error without managing its cause, i.e., nullable types and Arrow `Option` types".